docs/platforms/native/advanced-usage/signal-handling/index.mdx

```diff
@@ -0,0 +1,62 @@
+---
+title: Signal Handling
+description: "Learn about error handling common to all POSIX platforms."
+sidebar_order: 1000
+---
+The Native SDK uses signal handling on all supported platforms. On Linux (and Android), it is the primary mechanism to detect errors, while on macOS and Windows, signal handlers detect specific errors that other mechanisms do not detect or in lightweight backends that don't support the lower-level error handling mechanisms (for instance `inproc` on macOS).
+
+A (POSIX) signal is a way for an operating system kernel to inform an application about an exceptional event and provide a mechanism for the application to react to that event. The operating system delivers this information by interrupting the current execution of the affected  application with a "handler" function that the application registered before.
```

Comment:
I can't entirely agree here because "exception" is an overloaded term; it either is too generic or has a technical meaning that refers to something very different from signals. 

I am okay with using different wording than "exceptional event", but "exception" is confusing given what most technical readers have in mind when reading that term. What about the following:

> A (POSIX) signal is a way for an operating system kernel to interrupt the execution of an application to deliver a message of urgency, often with the intent to terminate the application. The affected application can react to that interruption with a "handler" function it registered before.

Reply:
IMO we can keep "exceptional event" because it was used intentional as a more generic term

Reply:
ah, makes sense. I'm ok with leaving it as "exceptional event" then.